src/libopenrave/trajectory.cpp

```diff
@@ -41,8 +41,12 @@ void TrajectoryBase::serialize(std::ostream& O, int options) const
     if( GetReadableInterfaces().size() > 0 ) {
         xmlreaders::StreamXMLWriterPtr writer(new xmlreaders::StreamXMLWriter("readable"));
         FOREACHC(it, GetReadableInterfaces()) {
-            BaseXMLWriterPtr newwriter = writer->AddChild(it->first);
-            it->second->Serialize(newwriter,options);
+            // some readable are not xml readable and does not get serialized here
```

Comment:
@ziyan @bucketzxm the point here is that we serialize any Readlabe and save it as part of the trajectory. Doesn't have to be XML, why not serialize the other readables?

Reply:
Because we don't plan to read `JSONReadable` out of XML format file. For example, we don't deserialize some JSONReadable out of a Collada file.

I think our focus should be trying to improve the binary format of the trajectory. This particular change here is just to provide back-compatibility, i.e. things that were in XML will still work, but no new JSON gets added to XML file.

Reply:
If we plan to support serializing `A` to both XML and JSON, we should make it implement both `XMLReadable` and `JSONReadable`, and it can serialize/deserialize freely between XML and JSON.

Reply:
At the time of writing, this https://github.com/rdiankov/openrave/pull/744 was not merged yet. We will work on top of those changes to support `JSONReadable` in binary format.

Reply:
@bucketzxm has added support for json format in the binary trajectory format.